s4e-backend/src/test/java/pl/cyfronet/s4e/service/SearchServiceTest.java

```diff
@@ -0,0 +1,270 @@
+import pl.cyfronet.s4e.data.repository.SceneRepository;
+import pl.cyfronet.s4e.properties.GeoServerProperties;
+import pl.cyfronet.s4e.properties.S3Properties;
+import pl.cyfronet.s4e.util.GeometryUtil;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+
+@BasicTest
+@Slf4j
+public class SearchServiceTest {
+    @Autowired
+    private SceneRepository sceneRepository;
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private TestDbHelper testDbHelper;
+    @Autowired
+    private SearchService searchService;
+    @Autowired
+    private GeoServerProperties geoServerProperties;
+    @Autowired
+    private S3Properties s3Properties;
+    @Autowired
+    private GeometryUtil geom;
+    @Autowired
+    private ObjectMapper objectMapper;
+    private ZoneParameter tz;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        testDbHelper.clean();
+        tz = new ZoneParameter(ZoneId.of("Europe/Warsaw"));
+        //add product
+        Product product = productRepository.save(Product.builder()
+                .name("108m")
+                .displayName("108m")
+                .description("Obraz satelitarny Meteosat dla obszaru Europy w kanale 10.8 µm z zastosowanie maskowanej palety barw dla obszarów mórz i lądów.")
+                .layerName("108m")
+                .build());
+        //addscenewithmetadata
+        val timestamp = LocalDateTime.now();
+        val s3Path = DateTimeFormatter.ofPattern("yyyyMMddHHmm").format(timestamp) + "Merkator_Europa_ir_108_setvak.tif";
+        val granulePath = geoServerProperties.getEndpoint() + "://" + s3Properties.getBucket() + "/" + s3Path;
+        Geometry footprint = geom.parseWKT("POLYGON((0 -10379291.247,0 0,10463673.528 0,10463673.528 -10379291.247,0 -10379291.247))", GeometryUtil.FACTORY_3857);
+        String metadataContent = "{\n" +
+                "   \"spacecraft\": \"Sentinel-1A\",\n" +
+                "   \"product_type\": \"GRDH\",\n" +
+                "   \"sensor_mode\": \"IW\",\n" +
+                "   \"processing_level\": \"1LD\",\n" +
+                "   \"cloud_cover\": 1.5,\n" +
+                "   \"polarisation\": \"Dual VV/VH\",\n" +
+                "   \"sensing_time\": \"2019-11-10T05:07:42.047432+00:00\",\n" +
+                "   \"ingestion_time\": \"2019-11-10T05:34:27.000000+00:00\",\n" +
+                "   \"relative_orbit_number\": \"22\",\n" +
+                "   \"polygon\": \"55.975475,19.579060 56.395580,15.414984 57.887905,15.835841 57.462494,20.167494\",\n" +
+                "   \"format\": \"GeoTiff\",\n" +
+                "   \"schema\": \"Sentinel-1.metadata.v1.json\"\n" +
+                "}";
```

Comment:
Possibly load these contents from files?

Reply:
changed to generate in loop, lets keep this one in code